docker/quarto/Dockerfile

```diff
@@ -0,0 +1,62 @@
+ARG VAST_VERSION
+ARG QUARTO_VERSION
+ARG VAST_CONTAINER_REGISTRY
+
+FROM $VAST_CONTAINER_REGISTRY/tenzir/vast:$VAST_VERSION AS quarto-base
+ARG QUARTO_VERSION
```

Comment:
That's docker syntax, need to repeat it in each stage you use it in. Annoying, isn't it?

Reply:
Woah 🤯

Reply:
I'm still not understanding why it would show up twice though in the same Dockerfile. Is it really that bad?

Reply:
a small example is the best to illustrate the behaviour:
```
ARG HELLO

FROM debian

ARG YOLO

RUN echo $HELLO
RUN echo $YOLO
```

```
version: '3'
services:
  test:
    build:
      context: .
      args:
        - HELLO=world
        - YOLO=swag
 ```

```
> docker compose build --progress=plain --no-cache
#5 [2/3] RUN echo $HELLO
#5 0.315 
#5 DONE 0.3s

#6 [3/3] RUN echo swag
#6 0.477 swag
#6 DONE 0.5s
```

As you can see, the HELLO variable was empty withing the stage. Actually, the top level declaration of YOLO is not required, I thought it was. But I would still argue that it is valuable to have the list of all the arguments at the top of the Dockerfile: for the overhead it adds, I find that it helps the reader a great deal. I don't have a very strong opinion about it though, DRY also makes sense as a principle in that case.

Reply:
> But I would still argue that it is valuable to have the list of all the arguments at the top of the Dockerfile

But then you need to make sure you always change two locations. That seems brittle compared to having to edit just one. If this is the convention in Docker land, I lift my hat, but otherwise I'd argue for KISS.

Reply:
not sure about a strong convention, but it would make me sleep better tonight to leave it 🙂